src/ticket.c

```diff
@@ -373,14 +402,15 @@ int list_ticket(char **pdata, unsigned int *len)
 	*pdata = NULL;
 	*len = 0;
 
-	alloc = booth_conf->ticket_count * (BOOTH_NAME_LEN * 2 + 128);
+	alloc = booth_conf->ticket_count * (BOOTH_NAME_LEN * 2 + 128 + 16);
 	data = malloc(alloc);
 	if (!data)
 		return -ENOMEM;
 
 	cp = data;
 	foreach_ticket(i, tk) {
-		if (is_time_set(&tk->term_expires)) {
+		if (is_time_set(&tk->term_expires) && (tk->mode != TICKET_MODE_MANUAL)) {
```

Comment:
In general (i.e. not only here), I'd move the is_manual() predicate to the front of the if statement. That way you guarantee that no code is going to be executed unwanted.

Reply:
sure, good idea!